rails_app/app/jobs/load_currencies_history_job.rb

```diff
@@ -8,19 +8,17 @@ def perform **options
 
     puts options[:date_range]
 
-    Currency.transaction do
-      # update record list
-      result = FreecurrencyApi.call endpoint: "historical", options: { date_from: options[:date_range].first, date_to: options[:date_range].last }
-
-      unless result.failure?
-        result.data.each do |date, records|
-          records.each do |currency_code, ex_rate|
-            record = CurrencyRecord.find_or_initialize_by record_date: date.to_date, code: currency_code
-            record.update latest_exchange_rate: ex_rate
-          end
-        end
-      end
+    # update record list
+    result = FreecurrencyApi.call endpoint: "historical", options: { date_from: options[:date_range].first, date_to: options[:date_range].last }
 
+    unless result.failure?
+      # insert_all ignores conflicts by default (doesn't fail if there is a conflict)
+      CurrencyRecord.insert_all result.data.flat_map { |date, incoming_records|
```

Comment:
This could probably be improved by parsing in batches and doing multiple `insert_all` instead of all at once. That way, memory consumption shouldn't be an issue anymore.

Reply:
For the record: 

https://apidock.com/rails/ActiveRecord/Batches/find_in_batches

Reply:
Thank you for the feedback.
I'm struggling to understand how I would use `find_in_batches` here. My understanding is that `find_in_batches` makes multiple queries to the database and is used to read/modify relations in, well, batches 😅.

Also, I'm assuming that if we can handle the whole response in memory (currently < 1 MB) we can handle composing the SQL query. If that was not the case then batching directly from FreecurrencyAPI would be the best solution. Either by requesting only a couple of currencies at a time or a shorter date time range.

Reply:
You're right, I got confused and thought the records were being read from 
memory. Agree that if we can handle everything in memory, this shuold be ok.